### ISSUE TITLE
Cannot have optional fields in SQLite 

### ISSUE THREAD

**Issue by ktalebian** [typeorm/typeorm#10511]
### Issue description

Cannot have optional fields in SQLite 

### Expected Behavior

I have an entity as followed:

```ts
@Entity('address')
export class AddressEntity extends CrudEntity<AddressId> implements Address {
  @Column('varchar')
  @Required()
  reference: AddressReference;

  @IdColumn()
  @IdProperty()
  @Required()
  referenceId: Id;

  @Column()
  @Required()
  street1: string;

  @Column()
  street2?: string;

  @Column()
  street3?: string;

  @Column()
  @Required()
  city: string;

  @Column()
  @Required()
  state: string;

  @Column()
  @Required()
  code: string;

  @Column()
  @Required()
  country: string;
}
```

You can see that `street2` and `street3` are optional. When running unit tests, I use SQLite in-memory for the db using

```
{
    type: 'sqlite',
    synchronize: true,
    database: ':memory:',
    entities: [...Object.values(entities), ...Object.values(joinEntities)],
}
```

as my option. However, if I do not provide _all_ fields, I get an error when inserting a row:

```
QueryFailedError: SQLITE_CONSTRAINT: NOT NULL constraint failed: address.street2
```

My actual production code is using `mysql` and the optional fields are respected using this db.

### Actual Behavior

Optional fields should be respected in sqlite mode.

### Steps to reproduce

Code provided above

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |    macOs      |
| Node.js version     | 18 |
| Typescript version  | 5.2.2  |
| TypeORM version     | 0.3.17  |
| sqlite3 | 5.1.6 |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [X] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

**Comment by alenap93** [comment]
columnOptions nullable is false by default : https://typeorm.io/entities#column-options

**Comment by OSA413** [comment]
Yeah, you need to specify that the column is optional, otherwise it will be NOT NULL (as for simple types, I don't remember what happens with relations).
